### ISSUE TITLE
Searching in DE returns RS address if no street name given

### ISSUE THREAD

**Issue by cottton** [openstreetmap/Nominatim#1454]
Example call:

    // GET https://nominatim.openstreetmap.org/search
    // 'Accept' => 'application/json',
    // 'Accept-Charset' => 'utf-8',
    // 'Content-Type' => 'application/json',
    [
        "street"         => '10', // missing street name
        "city"           => 'Weißendorf',
        "postcode"       => '07950',
        "country_code"   => 'DE',
        'format'         => 'json',
        'addressdetails' => 1,
        'limit'          => 1
    ]

For w/e reason user did not entered the street name.
So the search goes out with i.e. street "10".
I definitely would not expect to "leave the country" if the street name is missing.
Almost everything is there - so shoulnt we end up in that city at least?

_(I took a random street number with the same bug-result. This happen to other numbers too. Some end in RS, some elsewhere except for what i would have expected.)_

**Comment by lonvia** [comment]
There is no such parameter as `country_code` in the search API. Maybe you meant `countrycodes`?

In any case, note that Nominatim is not able to find something equivalent to 'all housenumbers 10 in weissendorf'. That's because of the way the data is structured in the database. With `countrycodes` you do get at least results in Germany only.

NB: you get results in Belgrade because OSM unhelpfully has the old medival German name [Griechisch Weissenburg](https://www.openstreetmap.org/node/60571493).

**Comment by cottton** [comment]
I hope i did not mix something up - im using this api: https://nominatim.org/release-docs/develop/api/Search/
 `https://nominatim.openstreetmap.org/search`

The offsets used so far:

    street
    city
    postcode
    country_code
    format
    addressdetails
    limit

but now i see `country` and `postalcode` (in the upper description on the docu).

`country` would be the country name. So "Deutschland".
But i got the country_iso "DE".
**So i need `countrycodes` ?**  

`postalcode` is weird since we get `postcode` back in the response. 
Does it handle both - `postcode` and `postalcode`?

So i guess this would be the correct request?

    [
        "street"         => 'SomeStreet 10',
        "city"           => 'SomeCity',
        "postalcode"     => '12345',
        "countrycodes"   => 'DE',
        'format'         => 'json',
        'addressdetails' => 1,
        'limit'          => 1
    ]

**Comment by mtmail** [comment]
Yes, that works. Documentation lists countrycodes value as lowercase (de) but uppercase works, too.

> Does it handle both - postcode and postalcode?

`postalcode` in the request, `postcode` in the response.

**Comment by cottton** [comment]
Ok, thank you!
Case closed =)

**Comment by lonvia** [comment]
Just to add a bit further information: the difference between `countrycodes` and `country` is rather subtle.

`countrycodes` is a _filter_. That means when using it, you really only get back results in the given countries. It does expect 2-letter iso codes (case does not matter) and you may also give a comma-separated list of countries.

`country` is a _matcher_. That means that an address within the country is more likely to end up in the results. (In reality, the current implementation downgrades an address outside the country so much that it never shows up in the results. That may make `country` look like a filter but that's just the implementation, really.) The format is also different. There can be just one country, You may use the name or the iso code though, both works.

Long story short, if you know you have a 2-letter iso code, preferably use `countrycodes`. It's faster.
